docs/source/contributing.generator.rst

```diff
@@ -0,0 +1,255 @@
+    >>> import garak.generators.replicate
+    >>> 
+
+If all goes well, no errors will appear. If some turn up, try and address those.
+
+The next step is to instantiate the class. Let's try with that ``meta/llama-2-70b-chat`` model.
+
+.. code-block:: bash
+
+    >>> g = garak.generators.replicate.ReplicateGenerator("meta/llama-2-70b-chat")
+    🦜 loading generator: Replicate: meta/llama-2-70b-chat
+    Traceback (most recent call last):
+    File "<stdin>", line 1, in <module>
+    File "/home/lderczynski/dev/garak/garak/generators/replicate.py", line 42, in __init__
+        raise ValueError(
+    ValueError: 🛑 Put the Replicate API token in the REPLICATE_API_TOKEN environment variable (this was empty)
+                    e.g.: export REPLICATE_API_TOKEN="r8-123XXXXXXXXXXXX"
+
+Oh, that's right! No API key. Looks like the validation exception is working as intended. Let's set up that value (maybe quit the interpreter, add it using the helpful suggestion in the exception method, and load up Python again).
+
+.. code-block:: bash
+
+    $ export REPLICATE_API_TOKEN="r8-not-a-real-token"
+    $ python
+    Python 3.11.9 (main, Apr 19 2024, 16:48:06) [GCC 11.2.0] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    >>> import garak.generators.replicate
+    >>> g = garak.generators.replicate.ReplicateGenerator("meta/llama-2-70b-chat")
+    🦜 loading generator: Replicate: meta/llama-2-70b-chat
+    >>> 
+
+Excellent! Now let's try a test generation:
+
+.. code-block:: bash
+
+    $ python
+    Python 3.11.9 (main, Apr 19 2024, 16:48:06) [GCC 11.2.0] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    >>> import garak.generators.replicate
+    >>> g = garak.generators.replicate.ReplicateGenerator("meta/llama-2-70b-chat")
+    🦜 loading generator: Replicate: meta/llama-2-70b-chat
+    >>> g.generate("test prompt", generations_this_call=1)
+    [" Sure, I'm happy to help! Can you please provide an actual prompt or question you'd like me to assist with? I'll do my best to provide a helpful and informative response while adhering to the guidelines you've outlined."]
+    >>>
+
+Well, this looks promising.
+
+The next step is to try some integration tests - executing garak from the command line, accessing this generator. There are some pointers in :doc:`contributing`. You might need to execute garak by specifying it as a Python module, running the command from the garak root code directory. Things to test are:
+
+* Does the new generator appear in ``python -m garak --list_generators``?
+* Does the generator work with a test probe, via ``python -m garak -m replicate -n meta/llama-2-70b-chat -p test.Blank``?
+* Do the garak tests pass? ``python -m pytest tests/``
+
+
+If you want to see the full, live code for the Replicate garak generator, it's here: `garak/generators/replicate.py <https://github.com/leondz/garak/blob/main/garak/generators/replicate.py>`_ .
+
+Done!
```

Comment:
Consider adding a comment that any edge case behaviors in `__init__` or `_call_model` may warrant adding a testing class to mock and exercise those conditions.

Reply:
done in 163b4e5